lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -16,56 +18,71 @@ def make_logger(server_config):
 
 
 class TestUnpackTarballs:
-    tar_name = "A.tar.xz"
-    tarlist = [tar_name]
-    linksrc = "TO-UNPACK"
-
-    def mock_glob(self, path):
-        """Returns the Tarball list"""
-        return TestUnpackTarballs.tarlist
-
-    def mock_resolve(self: Path, strict: bool = False):
-        """
-        Return a "full" fake file name
-
-        Args:
-            strict: Not used in the mock
-
-        Returns:
-            new Path with fake resolution
-        """
-        return Path(self.tar_name)
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
 
     def test_tarball_symlinks(self, monkeypatch, server_config, make_logger, caplog):
         """Tests for an incorrect symlink which results in an unresolved location"""
 
-        monkeypatch.setattr(glob, "iglob", self.mock_glob)
-        obj = Unpack(server_config, self.linksrc, make_logger)
-        total, success = obj.unpack_tarball()
-        assert (
-            f"Tarball link, '{self.tar_name}', does not resolve to a real location"
-            in caplog.text
-        )
+        def mock_resolve(self: Path, strict: bool = False):
+            """
+            Return a "full" fake file name
+
+            Args:
+                strict: Not used in the mock
+
+            Returns:
+                new Path with fake resolution
+            """
+            raise FileNotFoundError
+
+        monkeypatch.setattr(glob, "iglob", lambda x: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", mock_resolve)
+        obj = Unpack(server_config, make_logger)
+        result = obj.unpack()
+        for tar in self.tarlist:
+            assert (
+                f"Tarball link, '{tar}', does not resolve to a real location"
+                in caplog.text
+            )
+        assert result.ntotal == 2
+        assert result.nsuccess == 0
 
     def test_tarball_file(self, monkeypatch, server_config, make_logger, caplog):
         """Show that when tarball path cannot be accessed it throws a
         FileNotFoundError Exception"""
 
-        expected_error_msg = f"[Errno 2] No such file or directory: '{self.tar_name}'"
-        monkeypatch.setattr(glob, "iglob", self.mock_glob)
-        monkeypatch.setattr(Path, "resolve", self.mock_resolve)
-        obj = Unpack(server_config, self.linksrc, make_logger)
-        with pytest.raises(FileNotFoundError) as exc:
-            total, success = obj.unpack_tarball()
-        assert expected_error_msg == str(exc.value)
+        def mock_md5sum(filepath: Path):
+            """
+            Mock of md5sum module
```

Comment:
s/module/function/

(In the reference at line 68, `pbench.server.utils` is a module; `md5sum` is a function referenced in that module.)